token-service/src/main/java/org/p2p/token/service/TokenServiceModule.kt

```diff
@@ -55,7 +55,7 @@ object TokenServiceModule : InjectionModule {
         factory<TokenPriceRepository> { TokenPriceRemoteRepository(get(), get()) }
 
         factoryOf(::TokenServiceMapper)
-        factoryOf(::TokenMetadataRemoteRepository) bind TokenMetadataRepository::class
+        singleOf(::TokenMetadataRemoteRepository) bind TokenMetadataRepository::class
```

Comment:
you sure it needs to be single?

Reply:
Yes, I have plans for this repo. 
For now, we are reducing calls to shared prefs.